File: 151002/Kragel/lab3/publisher/src/main/java/by/bsuir/publisher/model/Tweet.java
package by.bsuir.publisher.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "tbl_tweet")
@EntityListeners(AuditingEntityListener.class)
public class Tweet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "title", nullable = false, unique = true, length = 64)
    private String title;

    @Column(name = "content", nullable = false, length = 2048)
    private String content;

    @CreatedDate
    @Column(name = "created")
    private Instant created;

    @LastModifiedDate
    @Column(name = "modified")
    private Instant modified;

    @ManyToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    private Creator creator;

    @ManyToMany(mappedBy = "tweets", cascade = {CascadeType.DETACH, CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH})
    private List<Sticker> stickers = new ArrayList<>();
}

File: 151004/Sakovsky/src/entities/Tweet.ts
import { TweetRequestToCreate } from "src/dto/request/TweetRequestToCreate";
import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from "typeorm";
import { Author } from "./Author";

@Entity({name: 'tbl_tweets'})
export class Tweet {
    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    @ManyToOne(() => Author, (author) => author.id)
    authorId: number;

    @Column({unique: true})
    title: string;

    @Column()
    content: string;

    @Column({nullable: false, default: new Date().toISOString().slice(0, 19).replace("T", " ")})
    created: Date;

    @Column({nullable: true, default: null})
    modified: Date;

    setTweet(dto: TweetRequestToCreate){
        this.authorId = dto.authorId;
        this.content = dto.content;
        this.title = dto.title;
        this.created = new Date();//new Date().toISOString().slice(0, 19).replace("T", " ")
        this.modified = null;
    }
}